File: backend/src/utils/apiResponse.js
function sendSuccess(res, data, message = "OK", statusCode = 200) {
    return res.status(statusCode).json({
        success: true,
        data,
        ...(message ? { message } : {}),
    });
}

function sendError(res, message = "Something went wrong", statusCode = 500, error = undefined) {
    return res.status(statusCode).json({
        success: false,
        message,
        ...(error !== undefined ? { error } : {}),
    });
}

module.exports = {
    sendSuccess,
    sendError,
};

File: app/utils/api_response.py
from typing import Any, Dict, Optional


def success_response(data: Any, message: str = "OK") -> Dict[str, Any]:
    return {
        "success": True,
        "data": data,
        "message": message,
    }


def error_response(message: str, error: Optional[Any] = None) -> Dict[str, Any]:
    payload: Dict[str, Any] = {
        "success": False,
        "message": message,
    }
    if error is not None:
        payload["error"] = error
    return payload
